### PR TITLE
Transpiled code warning

### PR DESCRIPTION
**Pull request by DylanC**
Detect transpiled code by using a median of function name length and provide warning to the user. Also checks file for a source map comment.

### CODE REVIEW

File: injects/logger.js [hunk 1]
```diff
@@ -32,7 +56,17 @@ const out = encoder.pipe(
...
+let firedOnce = false;
+
 const hook = asyncHooks.createHook({
+  before () {
+    if (!firedOnce && process.mainModule && checkForTranspiledCode(process.mainModule.filename)) {
+      // Show warning to user
+      fs.writeSync(3, 'source_warning', null, 'utf8')
+      firedOnce = true
+    }
+  },
```
Comment (mcollina) [inline injects/logger.js]: Please don't add a before hook. This is going to slow down things even more :/.

File: index.js [hunk 2]
```diff
@@ -69,9 +69,15 @@ class ClinicBubbleprof extends events.EventEmitter {
...
 
-    if (this.detectPort) {
-      proc.stdio[3].once('data', data => this.emit('port', Number(data), proc, () => proc.stdio[3].destroy()))
-    }
+    proc.stdio[3].once('data', data => { 
+      if (this.detectPort) {
+        this.emit('port', Number(data), proc, () => proc.stdio[3].destroy())
+      }
+      else if (data.toString() === 'source_warning') {
+        this.emit('warning', 'Transpiled code is not supported')
```
Comment (mcollina) [inline index.js]:
```suggestion
        this.emit('warning', 'The code is transpiled, bubbleprof does not support source maps yet.')
```

**Comment by mcollina** [comment]
Also, this needs tests.

File: injects/logger.js (hunk 1 above)
Comment (DylanC) [reply injects/logger.js]: No problem. Is there a place I can add it where process.mainModule would be available?

File: index.js [hunk 3]
```diff
@@ -44,7 +44,6 @@ class ClinicBubbleprof extends events.EventEmitter {
 
     if (this.detectPort) {
       logArgs.push('-r', 'detect-port.js')
-      stdio.push('pipe')
```
Comment (mcollina) [inline index.js]: why this change?

Comment (DylanC) [reply index.js]: @mcollina - We are creating the pipe in the array instead. 
`const stdio = ['inherit', 'inherit', 'inherit', 'pipe']`

Both Port and Source Warning will be using stdio[3].

File: index.js [hunk 4]
```diff
@@ -69,9 +68,14 @@ class ClinicBubbleprof extends events.EventEmitter {
       env: Object.assign({}, process.env, customEnv)
     })
 
-    if (this.detectPort) {
-      proc.stdio[3].once('data', data => this.emit('port', Number(data), proc, () => proc.stdio[3].destroy()))
-    }
+    proc.stdio[3].once('data', data => {
+      if (this.detectPort) {
```
Comment (goto-bus-stop) [inline index.js]: we should make sure that these features work together: if a port number is received, emit the 'port' event, and if the source warning value is received, emit the 'warning' event. Right now, the source warning will be treated as a port number if you're also using --detect-port.

File: index.js [hunk 5]
```diff
@@ -69,9 +68,14 @@ class ClinicBubbleprof extends events.EventEmitter {
...
+        this.emit('port', Number(data), proc, () => proc.stdio[3].destroy())
+      } else if (data.toString() === 'source_warning') {
+        this.emit('warning', 'The code is transpiled, bubbleprof does not support source maps yet.')
+        proc.stdio[3].destroy()
```
Comment (goto-bus-stop) [inline index.js]: Destroying fd 3 signals to the app that it needs to quit. In the 'port' event, our callback function is called once the --on-port or --autocannon process exits, and the app should exit in response. For the source map warning, that is not the case: we can just keep going, the warning is to inform the user why the graph might be hard to read.

File: injects/logger.js (hunk 1 above)
Comment (DylanC) [reply injects/logger.js]: @mcollina - We will be resolving this by introducing a process.nextTick and putting the check in there.
